integration/mesos/src/main/java/tachyon/mesos/AttributeEvaluator.java

```diff
@@ -0,0 +1,299 @@
+      attrsMap.put(attr.getName(), attr.getText().getValue());
+    }
+  }
+
+  public AttributeEvaluator(List<Protos.Attribute> attributes) {
+    setAttributes(attributes);
+  }
+
+  public boolean matchAttributes(String constraints) {
+    Lexer lexer = new Lexer(constraints);
+    Parser parser = new Parser(lexer);
+    AttributeExpression expression = parser.build();
+    return expression.interpret();
+  }
+
+  private class Lexer {
+    private StreamTokenizer input;
+
+    private int symbol = NONE;
+    public static final int EOL = -3;
+    public static final int EOF = -2;
+    public static final int INVALID = -1;
+
+    public static final int NONE = 0;
+
+    public static final int OR = 1;
+    public static final int AND = 2;
+    public static final int NOT = 3;
+
+    public static final int TRUE = 4;
+    public static final int FALSE = 5;
+
+    public static final int LEFT = 6;
+    public static final int RIGHT = 7;
+
+    public Lexer(String constraints) {
+      Reader r = buildReader(constraints);
+      input = new StreamTokenizer(r);
+      resetSyntax();
+    }
+
+    private void resetSyntax() {
+      input.resetSyntax();
+      input.wordChars('0', 'z');
+      input.whitespaceChars('\u0000', ' ');
+      input.whitespaceChars('\n', '\t');
+
+      input.ordinaryChar('(');
+      input.ordinaryChar(')');
+      input.ordinaryChar('&');
+      input.ordinaryChar('|');
+      input.ordinaryChar('!');
+    }
+
+    private Reader buildReader(String arg) {
+      ByteArrayInputStream byteStream = new ByteArrayInputStream(arg.getBytes());
+      return new BufferedReader(new InputStreamReader(byteStream));
+    }
+
+    private boolean matchAttributes(String val) {
+      if (val == null) {
+        return false;
+      }
+
+      if (!val.contains(":")) {
+        return false;
+      }
+
+      String[] kv = val.split(":");
+      String attr = kv[0];
+      String text = kv[1];
+
+      return text.equals(attrsMap.get(attr));
+    }
+
+    public int nextSymbol() {
+      try {
+        switch (input.nextToken()) {
+          case StreamTokenizer.TT_EOL:
+            symbol = EOL;
+            break;
+          case StreamTokenizer.TT_EOF:
+            symbol = EOF;
+            break;
+          case StreamTokenizer.TT_WORD:
+            if (matchAttributes(input.sval)) {
+              symbol = TRUE;
+            } else {
+              symbol = FALSE;
+            }
+            break;
+          case '(':
+            symbol = LEFT;
+            break;
+          case ')':
+            symbol = RIGHT;
+            break;
+          case '&':
+            symbol = AND;
+            break;
+          case '|':
+            symbol = OR;
+            break;
+          case '!':
+            symbol = NOT;
+            break;
+          default:
+            symbol = INVALID;
+        }
+      } catch (IOException e) {
+        symbol = EOF;
+      }
+      return symbol;
+    }
+  }
+
+  private class Parser {
```

Comment:
@BrickXu  have you looked into any library that provides boolean expression parsing? I feel it's a well-known concept, so there might be existing parser for it.